sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -368,8 +368,7 @@ case class FileSourceScanExec(
     val bucketed =
       selectedPartitions.flatMap { p =>
         p.files.map { f =>
-          val hosts = getBlockHosts(getBlockLocations(f), 0, f.getLen)
```

Comment:
Would it make sense to split the refactoring changes into their own PR? It's hard to tell at a glance which parts of the change are refactoring and which are new V2 implementation.

Reply:
Yes that is a good idea.

Reply:
I'm afraid these refactors only make sense in this PR, for reusing code between v1 and v2.

Reply:
Why does this only make sense for this PR? It looks like this is a reasonable refactor that could be stand-alone.

Reply:
moving code need a reason. The reason here is to help us to reuse the code. But if we do it in another PR, what is the reason? It doesn't make the code more clear IMO.

Reply:
Better organization to support other changes like this one is the reason.

@jose-torres was right to point out that these changes are self-contained enough to go in a separate PR and @gengliangwang and I both agreed. Why make this commit larger than necessary?

Reply:
if we agree that a separated PR is self-contained as it can help this PR, I'm also OK with it.

Reply:
Yeah, I think the commit itself would be self-contained reorganization. The motivation is to refactor for this PR, which is okay.